// @doist/todoist-api-typescript/dist/utils/colors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getColor = exports.colors = exports.taupe = exports.gray = exports.charcoal = exports.salmon = exports.magenta = exports.lavender = exports.violet = exports.grape = exports.blue = exports.lightBlue = exports.skyBlue = exports.turquoise = exports.mintGreen = exports.green = exports.limeGreen = exports.oliveGreen = exports.yellow = exports.orange = exports.red = exports.berryRed = void 0;
exports.berryRed = { name: 'Berry Red', id: 30, value: '#b8255f' };
exports.red = { name: 'Red', id: 31, value: '#db4035' };
exports.orange = { name: 'Orange', id: 32, value: '#ff9933' };
exports.yellow = { name: 'Yellow', id: 33, value: '#fad000' };
exports.oliveGreen = { name: 'Olive Green', id: 34, value: '#afb83b' };
exports.limeGreen = { name: 'Lime Green', id: 35, value: '#7ecc49' };
exports.green = { name: 'Green', id: 36, value: '#299438' };
exports.mintGreen = { name: 'Mint Green', id: 37, value: '#6accbc' };
exports.turquoise = { name: 'Turquoise', id: 38, value: '#158fad' };
exports.skyBlue = { name: 'Sky Blue', id: 39, value: '#14aaf5' };
exports.lightBlue = { name: 'Light Blue', id: 40, value: '#96c3eb' };
exports.blue = { name: 'Blue', id: 41, value: '#4073ff' };
exports.grape = { name: 'Grape', id: 42, value: '#884dff' };
exports.violet = { name: 'Violet', id: 43, value: '#af38eb' };
exports.lavender = { name: 'Lavender', id: 44, value: '#eb96eb' };
exports.magenta = { name: 'Magenta', id: 45, value: '#e05194' };
exports.salmon = { name: 'Salmon', id: 46, value: '#ff8d85' };
exports.charcoal = { name: 'Charcoal', id: 47, value: '#808080' };
exports.gray = { name: 'Gray', id: 48, value: '#b8b8b8' };
exports.taupe = { name: 'Taupe', id: 49, value: '#ccac93' };
exports.colors = [
    exports.berryRed,
    exports.red,
    exports.orange,
    exports.yellow,
    exports.oliveGreen,
    exports.limeGreen,
    exports.green,
    exports.mintGreen,
    exports.turquoise,
    exports.skyBlue,
    exports.lightBlue,
    exports.blue,
    exports.grape,
    exports.violet,
    exports.lavender,
    exports.magenta,
    exports.salmon,
    exports.charcoal,
    exports.gray,
    exports.taupe,
];
function getColor(colorId) {
    var color = exports.colors.find(function (color) { return color.id === colorId; });
    return color !== null && color !== void 0 ? color : exports.charcoal;
}
exports.getColor = getColor;
