// @doist/todoist-api-typescript/dist/types/entities.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = exports.Comment = exports.Attachment = exports.Label = exports.Section = exports.Project = exports.Task = exports.DueDate = exports.Int = void 0;
var runtypes_1 = require("runtypes");
exports.Int = runtypes_1.Number.withConstraint(function (n) { return Number.isInteger(n) || n + " is not a valid entity id. Should be an integer"; });
exports.DueDate = (0, runtypes_1.Record)({
    recurring: runtypes_1.Boolean,
    string: runtypes_1.String,
    date: runtypes_1.String,
}).And((0, runtypes_1.Partial)({
    datetime: runtypes_1.String,
    timezone: runtypes_1.String,
}));
exports.Task = (0, runtypes_1.Record)({
    id: exports.Int,
    order: exports.Int,
    content: runtypes_1.String,
    description: runtypes_1.String,
    projectId: exports.Int,
    sectionId: exports.Int,
    completed: runtypes_1.Boolean,
    labelIds: (0, runtypes_1.Array)(exports.Int),
    priority: exports.Int,
    commentCount: exports.Int,
    created: runtypes_1.String,
    url: runtypes_1.String,
}).And((0, runtypes_1.Partial)({
    parentId: exports.Int,
    due: exports.DueDate,
    assignee: exports.Int,
}));
exports.Project = (0, runtypes_1.Record)({
    id: exports.Int,
    name: runtypes_1.String,
    color: exports.Int,
    commentCount: exports.Int,
    shared: runtypes_1.Boolean,
    favorite: runtypes_1.Boolean,
    url: runtypes_1.String,
}).And((0, runtypes_1.Partial)({
    parentId: exports.Int,
    order: exports.Int,
    inboxProject: runtypes_1.Boolean,
    teamInbox: runtypes_1.Boolean,
    syncId: exports.Int,
}));
exports.Section = (0, runtypes_1.Record)({
    id: exports.Int,
    order: exports.Int,
    name: runtypes_1.String,
    projectId: exports.Int,
});
exports.Label = (0, runtypes_1.Record)({
    id: exports.Int,
    order: exports.Int,
    name: runtypes_1.String,
    color: exports.Int,
    favorite: runtypes_1.Boolean,
});
exports.Attachment = (0, runtypes_1.Record)({
    resourceType: runtypes_1.String,
}).And((0, runtypes_1.Partial)({
    fileName: runtypes_1.String,
    fileSize: exports.Int,
    fileType: runtypes_1.String,
    fileUrl: runtypes_1.String,
    fileDuration: exports.Int,
    uploadState: (0, runtypes_1.Union)((0, runtypes_1.Literal)('pending'), (0, runtypes_1.Literal)('completed')),
    image: runtypes_1.String,
    imageWidth: exports.Int,
    imageHeight: exports.Int,
    url: runtypes_1.String,
    title: runtypes_1.String,
}));
exports.Comment = (0, runtypes_1.Record)({
    id: exports.Int,
    content: runtypes_1.String,
    posted: runtypes_1.String,
}).And((0, runtypes_1.Partial)({
    taskId: exports.Int,
    projectId: exports.Int,
    attachment: exports.Attachment,
}));
exports.User = (0, runtypes_1.Record)({
    id: exports.Int,
    name: runtypes_1.String,
    email: runtypes_1.String,
});
